src/Feature/Hover/Feature_Hover.re

```diff
@@ -276,15 +289,18 @@ module View = {
         style={Styles.diagnostic(~theme=colorTheme)}
       />;
     };
-
-    <Oni_Components.HoverView x y theme=colorTheme>
-      {List.map(markdown => <hoverMarkdown markdown />, model.contents)
-       |> React.listToElement}
-      {model.contents != [] && diagnostic != []
-         ? <horizontalRule theme=colorTheme /> : React.empty}
-      {List.map(diag => <hoverDiagnostic diagnostic=diag />, diagnostic)
-       |> React.listToElement}
-    </Oni_Components.HoverView>;
+    switch (model.editor) {
+    | Some(editor') when editor' === editor =>
```

Comment:
You should generally avoid using physical equality (`===`) as it can easily lead to bugs, especially in functional programming where data structured tend to be replaced rather than mutated. Instead just compare the ids.

Reply:
Thanks, that's a much better idea. I was getting the `compare functional values` crash and lazily decided to use `===`. Thanks for correcting me!